src/cgutils.cpp

```diff
@@ -853,6 +853,31 @@ static Value *emit_bounds_check(const jl_cgval_t &ainfo, jl_value_t *ty, Value *
     return im1;
 }
 
+static void emit_inexacterror_unless(Value *cond, jl_value_t *type, jl_value_t *x, jl_codectx_t *ctx)
+{
+    BasicBlock *failBB = BasicBlock::Create(jl_LLVMContext,"fail",ctx->f);
+    BasicBlock *passBB = BasicBlock::Create(jl_LLVMContext,"pass");
+    builder.CreateCondBr(cond, passBB, failBB);
+    builder.SetInsertPoint(failBB);
+    Value *fname_val = stringConstPtr(ctx->funcName);
+#if JL_LLVM_VERSION >= 30700
+    builder.CreateCall(prepare_call(jlinexacterror_func),
+                       { fname_val, literal_pointer_val(type),
+                         boxed(x, ctx, false)});
+#else
+    builder.CreateCall3(prepare_call(jlinexacterror_func),
+                        fname_val, literal_pointer_val(type),
+                        boxed(x, ctx, false));
```

Comment:
I mean instead of using the `Value *auto_unbox(jl_value_t *x, jl_codectx_t *ctx)` which emits the expression and give you an unboxed value, you should "inline" that function, i.e. do
```cpp
    jl_cgval_t v = emit_expr(x, ctx);
    return auto_unbox(v, ctx);
```
to first emit that expression, getting the `jl_cgval_t` and then use the other signature of `auto_unbox` to get the `Value*` that's used in the original version of the code.